spec/controllers/media_objects_controller_spec.rb

```diff
@@ -279,6 +279,38 @@
           expect(new_media_object.other_identifier.find {|id_pair| id_pair[:source] == 'other'}).not_to be nil
           expect(new_media_object.other_identifier.find {|id_pair| id_pair[:source] == 'other'}[:id]).to eq('12345')
         end
+        it "should create a new media_object using ingest_api_hash of existing media_object" do
+          # master_file_obj = FactoryGirl.create(:master_file, master_file.slice(:files))
+          media_object = FactoryGirl.create(:fully_searchable_media_object)
+          master_file = FactoryGirl.create(:master_file, :with_derivative, :with_thumbnail, :with_poster, :with_structure, :with_captions)
+          media_object.ordered_master_files += [master_file]
```

Comment:
I think you can avoid this and the call to `update_dependent_properties!` by doing:
`FactoryGirl.create(:fully_searchable_media_object, master_files: [master_file])`

Reply:
I originally tried it that way, but something about it wasn't setting everything up correctly. I can't remember the details. My guess is the ordered_aggregate thing broke down.